docs/developer-guide.md

```diff
@@ -178,6 +178,55 @@ Mocks for unit tests rely on method signatures. Refresh them after interface cha
 make mocks
 ```
 
+---
+
+## Building a Custom Computation Management Server
+
+To integrate with CoCos agents, implement a gRPC server using [cvms.proto](https://github.com/ultravioletrs/cocos/blob/main/agent/cvms/cvms.proto):
+
+```proto
+rpc Process(stream ClientStreamMessage) returns (stream ServerStreamMessage);
+```
+
+Key server-side messages to send:
+
+- `ComputationRunReq`, `RunReqChunks`
+- `StopComputation`, `AgentStateReq`, `DisconnectReq`
+
+Expected agent responses:
+
+- `RunResponse`, `AgentLog`, `AgentEvent`
+- `AttestationResponse`, `StopComputationResponse`
+
+Example handler:
+
+```go
+func (s *server) Process(stream cvms.Service_ProcessServer) error {
+  for {
+    msg, err := stream.Recv()
+    if err != nil {
+      return err
+    }
+
+    switch m := msg.Message.(type) {
+    case *cvms.ClientStreamMessage_RunRes:
+      handleRunResponse(m.RunRes)
+    // handle other types...
+    }
+
+    _ = stream.Send(&cvms.ServerStreamMessage{
+      Message: &cvms.ServerStreamMessage_AgentStateReq{
+        AgentStateReq: &cvms.AgentStateReq{Id: "agent-1"},
+      },
+    })
+  }
+}
+```
+
+Use `RunReqChunks` to send large payloads, and validate attestation before running trusted workloads.
+
+---
```

Comment:
I feel like this is not sufficient, you'd rather talk about the requests and responses on the rpc